src/features/views/components/ViewColumnDialog/ColumnGallery/index.tsx

```diff
@@ -126,28 +129,34 @@ const ColumnGallery: FunctionComponent<ColumnGalleryProps> = ({
           width="20%"
         >
           <List>
-            {categories.map((category, index) => (
-              <ListItem
-                key={index}
-                onClick={() => {
-                  setSearchString('');
-                  setIsSearching(false);
-                  if (choiceContainerRef.current) {
-                    const element = choiceContainerRef.current.querySelector(
-                      `#category-${index}`
-                    );
-                    element?.scrollIntoView({ behavior: 'smooth' });
-                  }
-                }}
-                sx={{ cursor: 'pointer', paddingY: 2 }}
-              >
-                <Typography>
-                  <Msg
-                    id={`misc.views.columnDialog.categories.${category.key}.title`}
-                  />
-                </Typography>
-              </ListItem>
-            ))}
+            {categories.map((category, index) => {
+              const filteredKeys = filteringKeys(isRestrictedMode, category);
+              if (filteredKeys.length === 0) {
+                return;
+              }
```

Comment:
I feel like I would probably move this logic up above the JSX, to generate an array that contains only the necessary categories, and then use that array here instead of `categories`. Something like this (online 72 or something):

```ts
const filteredCategories = categories.filter((category) => !!filteringKeys(isRestrictedMode, category).length);
```

And then on line 132:

```ts
filteredCategories.map((category, index) => (
```

That way, you can get rid of the logic in this function, and make it a single-expression function again (less changes to the file).

Reply:
Thanks for feedback! I totally agree on your comments and it was really helpful! I tried to use git add -p this time and double checked the white spaces. Hope this is fine...!